go-controller/pkg/util/util.go

```diff
@@ -594,21 +594,24 @@ func IsUDNEnabledService(key string) bool {
 }
 
 // ServiceFromEndpointSlice returns the namespaced name of the service that corresponds to the given endpointSlice
-// in the given network
-func ServiceFromEndpointSlice(eps *discovery.EndpointSlice, netInfo NetInfo) (k8stypes.NamespacedName, error) {
+// in the given network. If the service label is missing the returned namespaced name and the error are nil.
+func ServiceFromEndpointSlice(eps *discovery.EndpointSlice, netInfo NetInfo) (*k8stypes.NamespacedName, error) {
 	labelKey := discovery.LabelServiceName
 	if netInfo.IsPrimaryNetwork() {
 		if eps.Labels[types.LabelUserDefinedEndpointSliceNetwork] != netInfo.GetNetworkName() {
```

Comment:
will this UDN network label always exist? I guess we don't care if its empty cause then it won't match ? IsEndpointSliceForNetwork util seems more accurate in the sense it handles the empty being assuming its default network...

Reply:
I see you do:
```
if util.IsNetworkSegmentationSupportEnabled() && !util.IsEndpointSliceForNetwork(newEpSlice, netInfo) {
		return nil
	}
```
before calling this util but yea maybe the util itself also needs to be foolproof to avoid putting burden on the caller?

Reply:
Since `ServiceFromEndpointSlice` takes the network as a parameter I think it should restrict the processing to this network and not return services that do not belong to it.